File: IotTemplates/WindowsIotCoreTemplates/ProjectTemplates/WiringAppCpp/StartupTask.cpp
using namespace Windows::ApplicationModel::Background;

// These functions should be defined in the sketch file
void setup();
void loop();

namespace $safeprojectname$
{
    [Windows::Foundation::Metadata::WebHostHidden]
    public ref class StartupTask sealed : public IBackgroundTask
    {
    public:
        virtual void Run(Windows::ApplicationModel::Background::IBackgroundTaskInstance^ taskInstance) 
        {
            auto deferral = taskInstance->GetDeferral();

            setup();
            while (true)
            {
                loop();
            }

            deferral->Complete();
        }
    };
}

File: Azure/HelloCloud.Headless/StartupTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net.Http;
using Windows.ApplicationModel.Background;
using System.Threading.Tasks;
using Microsoft.Azure.Devices.Client;
using System.Threading;

namespace HelloCloud
{
    public sealed partial class StartupTask : IBackgroundTask
    {
        public void Run(IBackgroundTaskInstance taskInstance)
        {
            SendDeviceToCloudMessagesAsync(taskInstance.GetDeferral());
        }

        static async void SendDeviceToCloudMessagesAsync(BackgroundTaskDeferral deferral)
        {
            var deviceClient = DeviceClient.Create(iotHubUri,
                    AuthenticationMethodFactory.
                        CreateAuthenticationWithRegistrySymmetricKey(deviceId, deviceKey),
                    TransportType.Http1);

            var str = "Hello, Cloud!";
            var message = new Message(Encoding.ASCII.GetBytes(str));

            await deviceClient.SendEventAsync(message);

            deferral.Complete();
        }
    }
}
